Core.Collectors/Collector/CollectorBase.cs

```diff
@@ -153,6 +153,14 @@ public async Task<bool> ProcessAsync(T collectionNode, long maxPageCount)
 
         private bool DetectLooping(IEnumerable<JObject> records)
         {
+            if (records.Count() == 0)
```

Comment:
NIT: I would move the `int recordCount = recordStrings.Count;` from line 170 to here and use it :)

Reply:
ah, no it's sth different, stupid me